vm/src/anystr.rs

```diff
@@ -440,4 +441,13 @@ where
             .map_err(|err| vm.new_value_error(err.to_string()))?
             .format(vm, values)
     }
+
+    fn py_bytes_cformat(&self, values: PyObjectRef, vm: &VirtualMachine) -> PyResult<String> {
+        let regex = Regex::new(r"%(?P<spec>[<>=^]?[+\- ]?\d*,?\.?\d*)s").unwrap();
+        let format_string = self.as_utf8_str().unwrap();
+        let format_byte_string = regex.replace_all(format_string, "%${spec}b").to_string();
+        CFormatString::from_str(&format_byte_string)
```

Comment:
Maybe instead of running a regex on the format string, you could have a parameter [here](https://github.com/RustPython/RustPython/blob/249f1218965472ad0c781010b65366c6d423eb4e/vm/src/cformat.rs#L319-L323) like `bytes_format: bool` that if true makes `CFormatPreconversor::Str` act like `CFormatPreconversor::Bytes`?

Reply:
That's the way I first tried. But I found that approach needs to change all of CFormatSpec and CFormatString when I pass bytes_format as parameter. I don't want to change unrelated working code for this change.

I realized why a regex is not recommended in this code, I need other good approaches.

Do I need to use new traits for that structs? I am waiting your advices.

Reply:
Sorry, I am working on CFormatSpec. It is nearly done but I did not push the PR yet.
I am going to push it now.
My apologize I should push it early.

Reply:
#2318